Track calypso_wooexpress_one_dollar_offer when a user clicks on My Ho…

moon0326 opened the pull request:
…me and has an active intro offer

### Changes proposed in this Pull Request:

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

Closes #1322  .

<!-- The next section is mandatory. If your PR doesn't require testing, please indicate that you are purposefully omitting instructions. -->

- [ ] This PR is a very minor change/addition and does not require testing instructions (if checked you can ignore/remove the next section).

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Otherwise, please include detailed instructions on how these changes can be tested. Please, make sure to review and follow the guide for writing high-quality testing instructions below. -->

- [ ] Have you followed the [Writing high-quality testing instructions guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions)?

1. Sign up for a new trial site.
2. Make your trial site SSH-able by following peapX7-1D4-p2
3. Clone this branch and run `npm run start` to build the changes.
4. SSH into your trial site and copy over `/build` directory to `~/htdocs/wp-content/mu-plugins/wpcomsh/vendor/automattic/wc-calypso-bridge/build`
5. Access the site and open browser inspector then paste `window.localStorage.setItem( 'debug', 'calypso:analytics' )`
6. Change inspector log level to verbose
7. Go to WooCommerce Home.
8. Confirm `calypso_wooexpress_one_dollar_offer` track has been fired.

Note: This won't work if a user clicks on `My Home` from a page where the `wcCalypsoBridge` JS variable is not available, such as on the My Mailboxes or Upgrades pages.

<!-- End testing instructions -->

### Other information:

-   [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?

<!-- Mark completed items with an [x] -->

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

Commit:
Track calypso_wooexpress_one_dollar_offer when a user clicks on My Home and has an active intro offer

ilyasfoo reviewed (changes requested):
Thanks for working on this, @moon0326!

Sorry this wasn't clear in the issue, we need the track to be fired when users `views` the homescreen. This is because we need to use the track to be fired as soon as the user comes from free trial setup.

Commit:
Update src/track-menu-item.ts

Co-authored-by: Ilyas Foo <foo.ilyas@gmail.com>

Commit:
Update src/track-menu-item.ts

Co-authored-by: Ilyas Foo <foo.ilyas@gmail.com>

moon0326 commented:
> Thanks for working on this, @moon0326!
> 
> Sorry this wasn't clear in the issue, we need the track to be fired when users `views` the homescreen. This is because we need to use the track to be fired as soon as the user comes from free trial setup.

Thanks for the clarification. We probably need a different approach in that case. I'll make an update soon.

Commit:
Remove home click track

Commit:
Record calypso_wooexpress_one_dollar_offer when the intro banner is shown for the first time

moon0326 commented:
> Thanks for working on this, @moon0326!
> 
> Sorry this wasn't clear in the issue, we need the track to be fired when users `views` the homescreen. This is because we need to use the track to be fired as soon as the user comes from free trial setup.

I've decided to call the track in the intro offer banner as it is only displayed when the offer is available. 

Updated in https://github.com/Automattic/wc-calypso-bridge/pull/1398/commits/35998fa912e3a940377bd2f2b9cdb8290998b5f8

I've also updated test introduction.

ilyasfoo reviewed (changes requested):
> Thanks for the clarification. We may need to take a different approach in that case. I'll make an update soon.
>
> Do we want to fire the track only once?

Good question, I think we should track it every time, since that's the behaviour for $1 offer track for plans and checkout screen.

> I've decided to call the track in the intro offer banner as it is only displayed when the offer is available.
>
> Updated in https://github.com/Automattic/wc-calypso-bridge/commit/35998fa912e3a940377bd2f2b9cdb8290998b5f8

I think we should make the behavior consistent with other $1 tracks, so we should track the view, regardless of banner status 🙏 

Also, please update changelog, thanks! 😊

psealock commented:
>Change inspector log level to verbose

Can you remind me how to do this again?

Commit:
Track calypso_wooexpress_one_dollar_offer on every page load

Commit:
Add changelog

moon0326 commented:
> > Change inspector log level to verbose
> 
> Can you remind me how to do this again?

Sure thing! 

![Screen Shot 2024-01-08 at 4 03 08 PM](https://github.com/Automattic/wc-calypso-bridge/assets/4723145/e9b131f8-74c1-44f0-a1c7-c05be2ddc732)

Commit:
Merge branch 'master' into add/1322-homescreen-track

ilyasfoo reviewed (changes requested):
Thanks for making the changes, @moon0326!

It seems like the SlotFill is being rendered in all wc-admin pages 😓
 
<img width="1506" alt="image" src="https://github.com/Automattic/wc-calypso-bridge/assets/3747241/2b55dcf8-3d8c-4c95-8559-a28c72fcdf4b">

I think we'll also need to check the admin path to make sure it's homescreen

moon0326 commented:
> Thanks for making the changes, @moon0326!
> 
> It seems like the SlotFill is being rendered in all wc-admin pages 😓
> 
> <img alt="image" width="1506" src="https://private-user-images.githubusercontent.com/3747241/295086525-2b55dcf8-3d8c-4c95-8559-a28c72fcdf4b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ4MzQwNjEsIm5iZiI6MTcwNDgzMzc2MSwicGF0aCI6Ii8zNzQ3MjQxLzI5NTA4NjUyNS0yYjU1ZGNmOC0zZDhjLTRjOTUtODU1OS1hMjhjNzJmY2RmNGIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDEwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAxMDlUMjA1NjAxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YmM0ZjVmZTFhYWNlNWRiNjE0ZjY2YWVhNzhlODMwYWM5Y2JlNTY4NWE3MDg4ZTVlNjIyN2NiZTFmZjBlMTdkOSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.hcUWeIdwFtZyI89xu5gJCJXtWcLT9WuR9SdyTZyJzII">
> I think we'll also need to check the admin path to make sure it's homescreen

Thank you for catching it! That's unexpected for sure. I thought we only render the banner on homescreen :( I'll make a fix and look into why the component is being rendered on other pages later.

Commit:
Make sure home banners are rendered only on Woo Home

Commit:
Merge branch 'master' into add/1322-homescreen-track

ilyasfoo commented on src/index.js:
```diff
@@ -146,16 +146,18 @@ if ( !! window.wcCalypsoBridge.isEcommercePlanTrial ) {
-			render: CalypsoBridgeHomescreenBanner,
-			scope: 'woocommerce-admin',
-		} );
+	if ( window.location.search === '?page=wc-admin' ) {
```
@moon0326 I think this is not reliable since there might be other URL parameters used.

I'd suggest to use `getPath` from `@woocommerce/navigation` where it's more reliable

moon0326 replied:
@ilyasfoo getPath() returns `location.pathname` and it's hard to determine Woo Home with the returned value AFAIK. Do you see situations where this code won't work?

moon0326 replied:
FYI, I've also tried using [determine_current_page](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/Admin/PageController.php#L119) and [get_current_page](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/Admin/PageController.php#L219) from PageController class. Both report `homescreen` when you're on a task page.

ilyasfoo replied:
Thanks, @moon0326! Unfortunately, I found another problem with the way we conditionally load the SlotFill 😅 

It seems if we navigate to another page first, such as `/wp-admin/admin.php?page=wc-admin&path=%2Fanalytics%2Foverview` and then navigate to homescreen using the menu bar, no banner will be loaded.

Sorry for going round and round on this, but let's not check for homescreen in the `registerPlugin` layer, but instead in the component before firing the tracks.

moon0326 replied:
> Sorry for going round and round on this, but let's not check for homescreen in the registerPlugin layer, but instead in the component before firing the tracks.

No worries. Thank you for catching these cases 🤦 

I'll run a thorough test again.

moon0326 replied:
@ilyasfoo Fixed in https://github.com/Automattic/wc-calypso-bridge/pull/1398/commits/6bb67a11658027fa0989e3579f604333be43f61a 🙏

ilyasfoo reviewed (changes requested):
> Thank you for catching it! That's unexpected for sure. I thought we only render the banner on homescreen :( I'll make a fix and look into why the component is being rendered on other pages later.

@moon0326 Thanks! I have a minor comment on the URL parameter detection

Commit:
Use getScreenFromPath to determine homescreen

Commit:
Fix a bug where track does not get fired when users visit a different page first

ilyasfoo reviewed (approved):
Tested well, LGTM! 🚢

Commit:
Merge branch 'master' into add/1322-homescreen-track
